<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ERCP2v2Class</class>
 <widget class="QMainWindow" name="ERCP2v2Class">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1387</width>
    <height>881</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>ERCP2v2</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>data/n2t.jpg</normaloff>data/n2t.jpg</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="OpenglPanel" name="widget" native="true">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>20</y>
      <width>754</width>
      <height>754</height>
     </rect>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>30</y>
      <width>171</width>
      <height>241</height>
     </rect>
    </property>
    <property name="title">
     <string>Camera Transform</string>
    </property>
    <widget class="QDoubleSpinBox" name="doubleSpinBoxCamPosX">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>30</y>
       <width>91</width>
       <height>22</height>
      </rect>
     </property>
     <property name="decimals">
      <number>5</number>
     </property>
     <property name="minimum">
      <double>-10000.000000000000000</double>
     </property>
     <property name="maximum">
      <double>10000.000000000000000</double>
     </property>
    </widget>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>30</y>
       <width>21</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>X</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="doubleSpinBoxCamPosY">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>60</y>
       <width>91</width>
       <height>22</height>
      </rect>
     </property>
     <property name="decimals">
      <number>5</number>
     </property>
     <property name="minimum">
      <double>-10000.000000000000000</double>
     </property>
     <property name="maximum">
      <double>10000.000000000000000</double>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>60</y>
       <width>21</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Y</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="doubleSpinBoxCamPosZ">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>90</y>
       <width>91</width>
       <height>22</height>
      </rect>
     </property>
     <property name="decimals">
      <number>5</number>
     </property>
     <property name="minimum">
      <double>-10000.000000000000000</double>
     </property>
     <property name="maximum">
      <double>10000.000000000000000</double>
     </property>
    </widget>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>90</y>
       <width>21</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Z</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="doubleSpinBoxCamAngX">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>130</y>
       <width>91</width>
       <height>22</height>
      </rect>
     </property>
     <property name="decimals">
      <number>5</number>
     </property>
     <property name="minimum">
      <double>-10000.000000000000000</double>
     </property>
     <property name="maximum">
      <double>10000.000000000000000</double>
     </property>
    </widget>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>130</y>
       <width>21</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>RX</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>170</y>
       <width>21</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>RY</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="doubleSpinBoxCamAngY">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>170</y>
       <width>91</width>
       <height>22</height>
      </rect>
     </property>
     <property name="decimals">
      <number>5</number>
     </property>
     <property name="minimum">
      <double>-10000.000000000000000</double>
     </property>
     <property name="maximum">
      <double>10000.000000000000000</double>
     </property>
    </widget>
    <widget class="QLabel" name="label_6">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>200</y>
       <width>21</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>RZ</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="doubleSpinBoxCamAngZ">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>200</y>
       <width>91</width>
       <height>22</height>
      </rect>
     </property>
     <property name="decimals">
      <number>5</number>
     </property>
     <property name="minimum">
      <double>-10000.000000000000000</double>
     </property>
     <property name="maximum">
      <double>10000.000000000000000</double>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>290</y>
      <width>171</width>
      <height>241</height>
     </rect>
    </property>
    <property name="title">
     <string>Object Transform</string>
    </property>
    <widget class="QDoubleSpinBox" name="doubleSpinBoxObjPosX">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>30</y>
       <width>91</width>
       <height>22</height>
      </rect>
     </property>
     <property name="decimals">
      <number>5</number>
     </property>
     <property name="minimum">
      <double>-10000.000000000000000</double>
     </property>
     <property name="maximum">
      <double>10000.000000000000000</double>
     </property>
    </widget>
    <widget class="QLabel" name="label_7">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>30</y>
       <width>21</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>X</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="doubleSpinBoxObjPosY">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>60</y>
       <width>91</width>
       <height>22</height>
      </rect>
     </property>
     <property name="decimals">
      <number>5</number>
     </property>
     <property name="minimum">
      <double>-10000.000000000000000</double>
     </property>
     <property name="maximum">
      <double>10000.000000000000000</double>
     </property>
    </widget>
    <widget class="QLabel" name="label_8">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>60</y>
       <width>21</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Y</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="doubleSpinBoxObjPosZ">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>90</y>
       <width>91</width>
       <height>22</height>
      </rect>
     </property>
     <property name="decimals">
      <number>5</number>
     </property>
     <property name="minimum">
      <double>-10000.000000000000000</double>
     </property>
     <property name="maximum">
      <double>10000.000000000000000</double>
     </property>
    </widget>
    <widget class="QLabel" name="label_9">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>90</y>
       <width>21</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Z</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="doubleSpinBoxObjAngX">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>130</y>
       <width>91</width>
       <height>22</height>
      </rect>
     </property>
     <property name="decimals">
      <number>5</number>
     </property>
     <property name="minimum">
      <double>-10000.000000000000000</double>
     </property>
     <property name="maximum">
      <double>10000.000000000000000</double>
     </property>
    </widget>
    <widget class="QLabel" name="label_10">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>130</y>
       <width>21</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>RX</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_11">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>170</y>
       <width>21</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>RY</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="doubleSpinBoxObjAngY">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>170</y>
       <width>91</width>
       <height>22</height>
      </rect>
     </property>
     <property name="decimals">
      <number>5</number>
     </property>
     <property name="minimum">
      <double>-10000.000000000000000</double>
     </property>
     <property name="maximum">
      <double>10000.000000000000000</double>
     </property>
    </widget>
    <widget class="QLabel" name="label_12">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>200</y>
       <width>21</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>RZ</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="doubleSpinBoxObjAngZ">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>200</y>
       <width>91</width>
       <height>22</height>
      </rect>
     </property>
     <property name="correctionMode">
      <enum>QAbstractSpinBox::CorrectToPreviousValue</enum>
     </property>
     <property name="decimals">
      <number>5</number>
     </property>
     <property name="minimum">
      <double>-10000.000000000000000</double>
     </property>
     <property name="maximum">
      <double>10000.000000000000000</double>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>550</y>
      <width>171</width>
      <height>211</height>
     </rect>
    </property>
    <property name="title">
     <string>View Port</string>
    </property>
    <widget class="QDoubleSpinBox" name="doubleSpinBoxFovy">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>30</y>
       <width>91</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_13">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>31</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Fovy</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="doubleSpinBoxNear">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>60</y>
       <width>91</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <double>0.010000000000000</double>
     </property>
    </widget>
    <widget class="QLabel" name="label_14">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>60</y>
       <width>21</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Near</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="doubleSpinBoxFar">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>90</y>
       <width>91</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <double>10000.000000000000000</double>
     </property>
    </widget>
    <widget class="QLabel" name="label_15">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>90</y>
       <width>21</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Far</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_16">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>130</y>
       <width>31</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>width</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_17">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>170</y>
       <width>31</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>height</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBoxWidth">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>130</y>
       <width>91</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>3000</number>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBoxHeight">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>170</y>
       <width>91</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>3000</number>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="pushButton">
    <property name="geometry">
     <rect>
      <x>1190</x>
      <y>790</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Quit</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_4">
    <property name="geometry">
     <rect>
      <x>1010</x>
      <y>30</y>
      <width>101</width>
      <height>231</height>
     </rect>
    </property>
    <property name="title">
     <string>Camera Control</string>
    </property>
    <widget class="QPushButton" name="pushButtonResetCamera">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>81</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Reset Camera</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_5">
    <property name="geometry">
     <rect>
      <x>1010</x>
      <y>290</y>
      <width>101</width>
      <height>211</height>
     </rect>
    </property>
    <property name="title">
     <string>Object Control</string>
    </property>
    <widget class="QPushButton" name="pushButtonResetModelMatrix">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>75</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Reset Model</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_6">
    <property name="geometry">
     <rect>
      <x>1020</x>
      <y>550</y>
      <width>151</width>
      <height>191</height>
     </rect>
    </property>
    <property name="title">
     <string>Main Control</string>
    </property>
    <widget class="QPushButton" name="pushButtonInitialization">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>121</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Initialization</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButtonStartTracking">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>110</y>
       <width>121</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Start Tracking</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButtonReference">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>75</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Reference</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBox">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>30</y>
       <width>42</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>3</number>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButtonResetTracking">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>150</y>
       <width>121</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Reset Tracking</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_7">
    <property name="geometry">
     <rect>
      <x>1190</x>
      <y>560</y>
      <width>161</width>
      <height>181</height>
     </rect>
    </property>
    <property name="title">
     <string>Ultilities</string>
    </property>
    <widget class="QPushButton" name="pushButtonCalibration">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>141</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Camera Calibration</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBoxNpointsCalibration">
     <property name="geometry">
      <rect>
       <x>91</x>
       <y>50</y>
       <width>51</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <number>5</number>
     </property>
     <property name="singleStep">
      <number>10</number>
     </property>
     <property name="value">
      <number>10</number>
     </property>
    </widget>
    <widget class="QLabel" name="label_18">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>50</y>
       <width>46</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>N Points</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButtonCalculateCalibration">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>80</y>
       <width>141</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Calculate</string>
     </property>
    </widget>
   </widget>
   <widget class="QTextEdit" name="textEditCommand">
    <property name="geometry">
     <rect>
      <x>1140</x>
      <y>30</y>
      <width>211</width>
      <height>231</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButtonTestOptimization">
    <property name="geometry">
     <rect>
      <x>1150</x>
      <y>300</y>
      <width>201</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Test Optimization</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1387</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>OpenglPanel</class>
   <extends>QWidget</extends>
   <header>openglpanel.h</header>
   <container>1</container>
   <slots>
    <signal>cameraChanged()</signal>
    <signal>finishSelectingPoints()</signal>
    <slot>startManualCalibration()</slot>
    <slot>setNumberOfPoints(int)</slot>
    <slot>calculateCalibration()</slot>
    <slot>saveRenderedImage()</slot>
    <slot>testOptimization()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="ercp2v2.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>pushButtonResetCamera</sender>
   <signal>clicked()</signal>
   <receiver>ERCP2v2Class</receiver>
   <slot>resetCamera()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1061</x>
     <y>97</y>
    </hint>
    <hint type="destinationlabel">
     <x>1161</x>
     <y>95</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonResetModelMatrix</sender>
   <signal>clicked()</signal>
   <receiver>ERCP2v2Class</receiver>
   <slot>resetModel()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1067</x>
     <y>352</y>
    </hint>
    <hint type="destinationlabel">
     <x>1199</x>
     <y>351</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>ERCP2v2Class</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1244</x>
     <y>832</y>
    </hint>
    <hint type="destinationlabel">
     <x>1291</x>
     <y>840</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonCalibration</sender>
   <signal>clicked()</signal>
   <receiver>widget</receiver>
   <slot>startManualCalibration()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1258</x>
     <y>620</y>
    </hint>
    <hint type="destinationlabel">
     <x>774</x>
     <y>567</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBoxNpointsCalibration</sender>
   <signal>valueChanged(int)</signal>
   <receiver>widget</receiver>
   <slot>setNumberOfPoints(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1311</x>
     <y>656</y>
    </hint>
    <hint type="destinationlabel">
     <x>740</x>
     <y>800</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>widget</sender>
   <signal>finishSelectingPoints()</signal>
   <receiver>ERCP2v2Class</receiver>
   <slot>enableCaculateCalibration()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>766</x>
     <y>478</y>
    </hint>
    <hint type="destinationlabel">
     <x>1216</x>
     <y>478</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonCalculateCalibration</sender>
   <signal>clicked()</signal>
   <receiver>ERCP2v2Class</receiver>
   <slot>pushButtonCalculateCalibrationClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1328</x>
     <y>684</y>
    </hint>
    <hint type="destinationlabel">
     <x>1329</x>
     <y>533</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonCalculateCalibration</sender>
   <signal>clicked()</signal>
   <receiver>widget</receiver>
   <slot>calculateCalibration()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1219</x>
     <y>685</y>
    </hint>
    <hint type="destinationlabel">
     <x>759</x>
     <y>724</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonTestOptimization</sender>
   <signal>clicked()</signal>
   <receiver>widget</receiver>
   <slot>testOptimization()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1280</x>
     <y>334</y>
    </hint>
    <hint type="destinationlabel">
     <x>779</x>
     <y>318</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>saveOpenGLImage()</signal>
  <slot>updateCameraUI()</slot>
  <slot>updateCameraPose()</slot>
  <slot>resetCamera()</slot>
  <slot>resetModel()</slot>
  <slot>objOrigin()</slot>
  <slot>enableCaculateCalibration()</slot>
  <slot>pushButtonCalculateCalibrationClicked()</slot>
  <slot>saveRenderedImage()</slot>
 </slots>
</ui>
